### ISSUE TITLE
Invitation to clients from partners to use the platform

### ISSUE THREAD

**Issue by ElenaBelyavskaya** [OpenMAVN/PM#94]
As a partner joined to the platform I want to invite my loyal customers to install the app

**AC's**
1.	Using QR code by partner
GIVEN I have a business registered on the platform (with KYC passed or not yet)
WHEN I open the details of the business
THEN I can find QR code and a direct installation URL for inviting customers to the platform on behalf of this business (the code/link should embed the partner id)
AND I can copy the url and Save QR code as an image
AND there are explanation tool-tips/hints ('This link/QR code will allow your customers to join the platform; send this link to your customers who don't have the mobile app yet/ Share this QR code to your customers who don't have the mobile app yet')
AND the QR code or the URL contains referral code which allows identifying new customers referred by this exact business (mobile app can fetch this referral code from AppStore/PlayMarket after its installation and then use during customer registration process)

2.	Accepting invitation by customers
2.1. GIVEN a client received URL with invitation to the platform from an SME 
WHEN the client uses the URL
THEN he is redirected to install the app (AppStore/Play Market)

2.2. GIVEN an SME has shared with a client the QR code with invitation to the platform
WHEN the client scans the QR code
THEN he is redirected to install the app (AppStore/Play Market)
 
3.	Registering a new customer with referral link to the inviting Business
GIVEN a customer has used a referral QR code OR url provided by a particular Business
WHEN the customer installs the app and signs up on the platform (after the email is confirmed)
THEN the link is recorded for this customer with the business from which this customer was invited (connection Customer id - Partner id)

**Comment by starkmsu** [comment]
@ElenaBelyavskaya , please describe the full flow - including AdminPortal, mobile app and AppStore/Playmarket

**Comment by ElenaBelyavskaya** [comment]
@starkmsu, added one more scenario, please review if it's enough info now

**Comment by starkmsu** [comment]
@ElenaBelyavskaya , almost there - we need to add a part about referral code that should be a part of mobile pp installation URL, so mobile app can fetch it from AppStore/PlayMarket after its installation to use during customer registration process.

**Comment by ElenaBelyavskaya** [comment]
@starkmsu by "referral code" do you mean the code of the partner in fact, so that we could record by whom the customer was invited?
If yes, I described it briefly by the last 3 rows of scenario 2... Or is it smth else?

**Comment by starkmsu** [comment]
@ElenaBelyavskaya , yes, we need to explicitly describe the flow of referral code: where it's generated, how it's added to URL/QR for SME, how mobile app get this code from AppStore/PlayMarket to pass into the registration flow, how backend should use this code from registration params

**Comment by ElenaBelyavskaya** [comment]
@starkmsu I have added an explanation to point 1 re the referral code, and also described creating a referral link between customer and business separately, in point 3.
Should you need other/deeper details, let's describe them in sub-tasks.

**Comment by vitaliidasaev** [comment]
> THEN I can find and copy QR code OR direct installation URL for inviting customers to the platform on behalf of this business (the code/link should embed the partner id)

@pddimitrova we need deeplink for installation url.
@starkmsu @ElenaBelyavskaya there is two ways: 
1) to put referralCode into url in admin-api
2) on FE

**Comment by starkmsu** [comment]
@vitaliidasaev , since FE is overloaded and BE has free hands, let's do installation URL generation on the BE side

**Comment by starkmsu** [comment]
@pddimitrova , please paste links here to our mobile app on both AppStore and Google Play

**Comment by ElenaBelyavskaya** [comment]
Didn't find invite QR code/url on admin, @vitaliidasaev do we have it?
